packages/flutter/test/material/elevated_button_test.dart

```diff
@@ -1914,6 +1914,52 @@ void main() {
     expect(controller.value, <MaterialState>{MaterialState.disabled});
     expect(count, 1);
   });
+
+  testWidgetsWithLeakTracking('ElevatedButton.icon honors iconAlignment', (WidgetTester tester) async {
+    final Key key = UniqueKey();
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Center(
+          child: ElevatedButton.icon(
+            key: key,
+            icon: const Icon(Icons.add),
+            onPressed: () {},
+            label: const Text('button'),
+          ),
+        ),
+      ),
+    );
+    final Row rowWidget = tester.widget<Row>(
+      find.descendant(
+        of: find.byKey(key),
+        matching: find.byType(Row),
+      ),
+    );
+    expect(rowWidget.children.first.runtimeType, Icon);
+    expect(rowWidget.children.last.runtimeType, Flexible);
```

Comment:
```suggestion
    final Row rowWidget = tester.widget<Row>(
      find.descendant(
        of: find.byKey(key),
        matching: find.byType(Row),
      ),
    );
    expect(rowWidget.children.first.runtimeType, Icon);
    expect(rowWidget.children.last.runtimeType, Flexible);
```

Rather than testing `runtimeType` in a Row. This should check for the icon itself and its position in the button.

Reply:
Actually list contains 3 elements: Icon/Text, Spacing, Text/Icon. Hence I used runtimeType approach.

Reply:
Will do it today

Reply:
was stuck on an unexpected mistake - https://github.com/flutter/flutter/pull/137348#discussion_r1436621698

will work on correct expect statements today.

Reply:
last action item on my plate, will wrap it today. sorry for the delay.

Reply:
Implemented